modules/broker-core/src/main/grammar/SelectorParser.jj

```diff
@@ -87,47 +87,241 @@ TOKEN [IGNORE_CASE] :
   | <  ESCAPE  : "ESCAPE">
   | <  IN      : "IN">
   | <  IS      : "IS">
-  | <  TRUE    : "TRUE" >
-  | <  FALSE   : "FALSE" >
   | <  NULL    : "NULL" >
 }
 
 TOKEN [IGNORE_CASE] :
 {
-    <NUMERIC_LITERAL: (["0"-"9"])+(["L"])?>
-   | <STRING_LITERAL: "'" (("''") | ~["'"] )* "'">
+    < DECIMAL_LITERAL: ["1"-"9"] (["0"-"9"])* (["l","L"])? >
+     | < HEX_LITERAL: "0" ["x","X"] (["0"-"9","a"-"f","A"-"F"])+ >
+   | < OCTAL_LITERAL: "0" (["0"-"7"])* >
+   | < STRING_LITERAL: "'" ( ("''") | ~["'"] )*  "'" >
+   | < FLOATING_POINT_LITERAL:
+             (["0"-"9"])+ "." (["0"-"9"])* (<EXPONENT>)? // matches: 5.5 or 5. or 5.5E10 or 5.E10
+           | "." (["0"-"9"])+ (<EXPONENT>)?              // matches: .5 or .5E10
+           | (["0"-"9"])+ <EXPONENT>                     // matches: 5E10
+       >
+     | < #EXPONENT: "E" (["+","-"])? (["0"-"9"])+ >
 }
-
 TOKEN :
 {
     <IDENTIFIER : ["a"-"z","A"-"Z"](["a"-"z","A"-"Z","0"-"9","_","$"])*>
 }
+TOKEN :
+    {
+     < EQUAL: "=" >
+    | < GTREATER: ">" >
+    | < LESS: "<" >
+    | < GTREATER_EQAL: ">=" >
+    | < LESS_EQUAL: "<=" >
+    | < NOTEQUAL: "<>" >
+    }
 
 BooleanExpression parseSelector() :
 { BooleanExpression value; }
 {
-     value=ComparisonExpression() <EOF>
+     value= OrExpression() <EOF>
```

Comment:
Shall we add a space before and after the '=' sign? Shall we correct this in all places?